### PR TITLE
[PBM-906] add ensure-oplog command to save missed oplog chunks

### PR DESCRIPTION
**Pull request by defbin**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| New cluster psmdb 4.4 | passed | - |
| New cluster psmdb 4.2 | passed | - |
| New cluster psmdb 5.0 | failed | - |
| Non-sharded psmdb 4.2 | passed | failed |
| Non-sharded psmdb 4.4 | passed | passed |
| Single-node psmdb 4.2 | passed | failed |
| Single-node psmdb 4.4 | failed | failed |
| Non-sharded psmdb 5.0 | passed | failed |
| Sharded psmdb 4.2 | failed | failed |
| Sharded psmdb 4.4 | failed | failed |
| Sharded psmdb 5.0 | failed | passed |
| Single-node psmdb 5.0 | failed | failed |

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| New cluster psmdb 4.4 | passed | - |
| New cluster psmdb 4.2 | passed | - |
| New cluster psmdb 5.0 | failed | - |
| Non-sharded psmdb 4.2 | passed | failed |
| Non-sharded psmdb 4.4 | failed | passed |
| Single-node psmdb 4.2 | passed | failed |
| Non-sharded psmdb 5.0 | failed | passed |
| Single-node psmdb 4.4 | passed | passed |
| Single-node psmdb 5.0 | passed | passed |
| Sharded psmdb 4.2 | passed | failed |
| Sharded psmdb 5.0 | failed | failed |
| Sharded psmdb 4.4 | failed | failed |

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| Sharded psmdb 4.2 | failed | passed |
| Sharded psmdb 4.4 | failed | passed |
| New cluster psmdb 4.2 | failed | - |
| Sharded psmdb 5.0 | failed | passed |
| New cluster psmdb 5.0 | passed | - |
| New cluster psmdb 4.4 | passed | - |
| Non-sharded psmdb 4.2 | failed | passed |
| Non-sharded psmdb 4.4 | failed | passed |
| Non-sharded psmdb 5.0 | failed | passed |
| Single-node psmdb 4.2 | failed | failed |
| Single-node psmdb 4.4 | failed | passed |
| Single-node psmdb 5.0 | failed | passed |

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| Sharded psmdb 4.2 | failed | passed |
| Sharded psmdb 5.0 | failed | failed |
| Sharded psmdb 4.4 | failed | passed |
| Single-node psmdb 4.4 | failed | failed |
| New cluster psmdb 4.2 | failed | - |
| New cluster psmdb 4.4 | failed | - |
| New cluster psmdb 5.0 | failed | - |
| Non-sharded psmdb 4.2 | failed | passed |
| Non-sharded psmdb 4.4 | failed | failed |
| Non-sharded psmdb 5.0 | failed | passed |
| Single-node psmdb 4.2 | failed | failed |
| Single-node psmdb 5.0 | failed | failed |

File: agent/oplog.go [hunk 1]
```diff
@@ -78,3 +88,185 @@ func (a *Agent) OplogReplay(r *pbm.ReplayCmd, opID pbm.OPID, ep pbm.Epoch) {
...
+			l.Error("release lock: %v", err)
+		}
+	}()
+
+	from, err := findPreviousOplogTS(a.pbm.Context(), a.node.Session(), r.From)
+	if err != nil {
+		if errors.Is(err, mongo.ErrNoDocuments) {
+			err = pbm.ErrNotFound
+		}
+		l.Error("lookup first oplog record: %s", err.Error())
```
[the remark this answers is not in the document]
Comment (defbin) [reply agent/oplog.go:141]: 🤷‍♂️

**Pull request merged**